cmd/utils_test.go

```diff
@@ -0,0 +1,113 @@
+package cmd
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/iancoleman/strcase"
+	"github.com/stretchr/testify/require"
+)
+
+func TestScaffoldTutorialNameservice(t *testing.T) {
+	tmp, err := ioutil.TempDir("", "scaffold-test-")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer os.RemoveAll(tmp) // clean up
+	os.Chdir(tmp)
+
+	user := "username"
+	repo := "repository"
+	dir := "nameservice"
+	nameRaw := dir
+	nameCapitalCamelCase := strcase.ToCamel(nameRaw)
+	nameLowerCamelCase := strcase.ToLowerCamel(nameRaw)
+	nameLowerCase := strings.ToLower(nameLowerCamelCase)
+	ns := UserRepoArgs{
+		Dir:                  dir,
+		User:                 user,
+		Repo:                 repo,
+		NameRaw:              nameRaw,
+		NameLowerCase:        nameLowerCase,
+		NameCapitalCamelCase: nameCapitalCamelCase,
+		NameLowerCamelCase:   nameLowerCamelCase,
+	}
+	err = scaffold(dir, outputPath, ns)
+	require.NoError(t, err)
+}
+func TestScaffoldTutorialHelloChain(t *testing.T) {
+	tmp, err := ioutil.TempDir("", "scaffold-test-")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer os.RemoveAll(tmp) // clean up
+	os.Chdir(tmp)
+
+	user := "username"
+	repo := "repository"
+	dir := "hellochain"
+	nameRaw := dir
+	nameCapitalCamelCase := strcase.ToCamel(nameRaw)
+	nameLowerCamelCase := strcase.ToLowerCamel(nameRaw)
+	nameLowerCase := strings.ToLower(nameLowerCamelCase)
+	ns := UserRepoArgs{
+		Dir:                  dir,
+		User:                 user,
+		Repo:                 repo,
+		NameRaw:              nameRaw,
+		NameLowerCase:        nameLowerCase,
+		NameCapitalCamelCase: nameCapitalCamelCase,
+		NameLowerCamelCase:   nameLowerCamelCase,
+	}
+	err = scaffold(dir, outputPath, ns)
+	require.NoError(t, err)
+}
+
+func TestScaffoldApp(t *testing.T) {
```

Comment:
can we also build/run the app here? Or is that better done in CI?

Reply:
we can do app level scaffolding but not module level, which i think is fine. I can spend time next week working on this

Reply:
I think app level is fine for now and adding module support to the testing at a later date.